### PR TITLE
feat: add support for mini-css-extract-plugin on demand

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
WIP

**If relevant, did you update the documentation?**
N/A

**Summary**
Adding support for `mini-css-extract-plugin` in default template.

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

**Comment by rishabh3112** [comment]
@webpack/cli-team need review

**Review by alexander-akait — COMMENTED** [review]
I think we need small improve here, we recommend to use `mini-css-extract-plugin` for production, ideas?

**Comment by rishabh3112** [comment]
> I think we need small improve here, we recommend to use mini-css-extract-plugin for production, ideas?

Yeah I have, we could make configuration as `function` and use `webpack-merge`. Will do in separate PR.

**Review by alexander-akait — COMMENTED** [review]
So in future we will improve like:
```
Do you want to extract CSS for every file?
- Yes
- Only for production
- No
```

**Comment by rishabh3112** [comment]
> So in future we will improve like:
> 
> ```
> Do you want to extract CSS for every file?
> - Yes
> - Only for production
> - No
> ```

Yup. Will open issue for it after this PR's merge.

**Comment by alexander-akait** [comment]
Let's put it here https://github.com/webpack/webpack-cli/issues/2430

**Review by alexander-akait — APPROVED** [review]
/cc @webpack/cli-team

**Review by snitin315 — APPROVED** [review]

**Review by anshumanv — APPROVED** [review]

**Comment by rishabh3112** [comment]
/cc @webpack/cli-team  I am unable to see logs for CI runs which failed. Can anyone see them?
image

**Comment by alexander-akait** [comment]
github actions is very buddy today on macos

**Comment by rishabh3112** [comment]
Again trying to run CI, maybe its fixed now 🤞🏽

File: packages/generators/src/handlers/default.ts [hunk 1]
```diff
@@ -82,6 +82,14 @@ export async function questions(self: CustomGenerator, Question: Record<string,
         self.force,
     );
 
+    const { isExtractPlugin } = await Question.Confirm(
+        self,
+        'isExtractPlugin',
+        'Do you want to extract CSS for every file?',
```
Comment (evenstensberg) [inline packages/generators/src/handlers/default.ts:88]: This is misleading for beginners, and we should phrase this differently. Maybe, `Do you want to split CSS into seperate files`?

Comment (alexander-akait) [reply packages/generators/src/handlers/default.ts:88]: we don't do split here, we extract :smile:

**Comment by rishabh3112** [comment]
Thanks

**Pull request merged**